packages/ragbits-core/src/ragbits/core/prompt/prompt.py

```diff
@@ -119,14 +130,17 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             self._render_template(self.system_prompt_template, input_data) if self.system_prompt_template else None
         )
         self.rendered_user_prompt = self._render_template(self.user_prompt_template, input_data)
+        self._image_input_type = InputType(collection_type=list, element_type=bytes)
```

Comment:
I wonder whether type `bytes` is enough to know that the field contains an image. For example, in the future if we allow attaching audio to prompts, most likely the audio data will also be provided as `bytes`.

I would either:
* Create [a custom Pydantic data type](https://docs.pydantic.dev/2.3/usage/types/custom/) for images, that would allow us to detect them unequivocally
* Leave automatic detection of images for now, and allow users to provide `image_fields` by hand

Reply:
done - currently we support definition of `image_fields` with class variables